### ISSUE TITLE
Improve logging for importing Roslyn issues

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#2920]
### Description

Currently the logs for importing results don't help a lot 

We don't have debug logs for listing the paths of the reports
```
2019-10-29T19:13:52.0274041Z 15:13:52.022 DEBUG: File '<PATH>' was already processed. Skip it
2019-10-29T19:13:52.0278253Z 15:13:52.022 INFO: Importing results from 6 proto files in 'F:\agent3\_work\12\.sonarqube\out\8\output-cs'
2019-10-29T19:13:55.9374427Z 15:13:55.928 INFO: Importing results from 6 proto files in 'F:\agent3\_work\12\.sonarqube\out\27\output-cs'
2019-10-29T19:13:56.1273117Z 15:13:56.116 DEBUG: Not enough content in '<PATH>' to have CPD blocks, it will not be part of the duplication detection
2019-10-29T19:13:56.9994282Z 15:13:56.991 INFO: Importing 10 Roslyn reports
2019-10-29T19:13:59.1531674Z 15:13:59.147 INFO: Sensor C# [csharp] (done) | time=80533ms
```

This ticket requires to add more debug/trace logs for listing
- when the issues are pushed to SQ (at least statistics - "Sent X results to SQ"?)
- more debug details when importing results from proto files, statistics even for happy paths
- how many issues have been reported in the roslyn reports
- anything else that would help SUPPORT-16265

**Comment by andrei-epure-sonarsource** [comment]
copying info from the duplicate issue #3629

The Roslyn analyzer is executed during the build by `msbuild`. Then it outputs the results in a JSON file in a path like `PROJECT\\bin\\Debug\\netcoreapp3.1\\TestOnlyProject.dll.RoslynCA.json` . This report gets picked up by the Java plugin during the `scanner-cli` execution (which happens in the END step of the Scanner for MSBuild execution).

The Java plugin ("C# Sensor") opens the Roslyn reports and parses them (see [RoslynDataImporter](https://github.com/SonarSource/sonar-dotnet/blob/490b497f682a984bd984f6e450a1a1ee61d16a22/sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/RoslynDataImporter.java#L54), `SarifParserFactory.java`, `SarifParser01And04.java` and `SarifParser10.java`).

Inside the SARIF parsers, the `SarifParserCallback` is passed, which then calls the scanner API to add issues to the SQ project.

What currently gets logged (necessary for support and diagnostics):
- each issue is logged in the `msbuild` logs by the Roslyn integration
- the `....RoslynCA.json`) file is present on disk and we could ask for it
- RoslynDataImporter logs how many Roslyn reports are imported (this can show if a project didn't have its roslyn report imported)

The logs we lack:
- how many issues from each report get read
- the total number of issues that are pushed to SQ

Having these logs can enable us figure out analysis problems, in case the analysis behaves differently in different environments. Also, if there are problems when pushing the data to SonarQube or SonarCloud (if we parse the correct number of issues but less issues show up in SQ/SC).

The logging statements can be added in the Sarif* classes. Probably we'll need some statistics collector to capture the data before logging it (we don't want to log a line for each issue getting read , we want statistics).

**Comment by pavel-mikula-sonarsource** [comment]
@andrei-epure-sonarsource I think this is already implemented by #4270 and we can close this.
